src/flexmock/api.py

```diff
@@ -392,77 +381,66 @@ def _handle_matched_expectation(
                 )
             expectation.times_called += 1
             expectation.verify(final=False)
-            _pass_thru = _getattr(expectation, "_pass_thru")
-            _replace_with = _getattr(expectation, "_replace_with")
-            if _pass_thru:
+            if expectation._pass_thru:
                 return pass_thru(expectation, runtime_self, *kargs, **kwargs)
-            if _replace_with:
-                return _replace_with(*kargs, **kwargs)
-            return_values = _getattr(expectation, "_return_values")
-            if return_values:
-                return_value = return_values[0]
-                del return_values[0]
-                return_values.append(return_value)
-            else:
-                return_value = ReturnValue()
-            if return_value.raises:
-                if inspect.isclass(return_value.raises):
-                    raise return_value.raises(
-                        *return_value.value["kargs"], **return_value.value["kwargs"]
-                    )
-                raise return_value.raises  # pylint: disable=raising-bad-type
-            return return_value.value
+            if expectation._replace_with:
+                return expectation._replace_with(*kargs, **kwargs)
+            return_values = expectation._return_values or [ReturnValue()]
+            return_value = return_values.pop(0)
+            return_values.append(return_value)
+            if not return_value.raises:
+                return return_value.value
+            if inspect.isclass(return_value.raises):
+                raise return_value.raises(
+                    *return_value.value["kargs"], **return_value.value["kwargs"]
+                )
+            raise return_value.raises  # pylint: disable=raising-bad-type
 
         def mock_method(runtime_self: Any, *kargs: Any, **kwargs: Any) -> Any:
             arguments = {"kargs": kargs, "kwargs": kwargs}
             expectation = FlexmockContainer.get_flexmock_expectation(self, name, arguments)
             if expectation:
                 return _handle_matched_expectation(expectation, runtime_self, *kargs, **kwargs)
             # inform the user which expectation(s) for the method were _not_ matched
-            expectations = [
-                expectation
+            error_msg = _format_args(name, arguments) + "\n".join(
+                "\nDid not match expectation %s" % _format_args(name, expectation._args)
```

Comment:
Since we are supporting starting 3.6, how about making a task to replace all those with f-strings ?

Reply:
https://github.com/flexmock/flexmock/pull/61